tests/test_reordering.f90

```diff
@@ -0,0 +1,180 @@
+program test_reorder
+    use iso_fortran_env, only: stderr => error_unit
+    use mpi
+
+    use m_allocator, only: allocator_t, field_t
+    use m_base_backend, only: base_backend_t
+    use m_solver, only: allocate_tdsops
+    use m_tdsops, only: dirps_t, tdsops_t
+
+    use m_common, only: dp, pi, globs_t, set_pprev_pnext, &
+                       RDR_X2Y, RDR_X2Z, RDR_Y2X, RDR_Y2Z, RDR_Z2X, RDR_Z2Y, dir_X, dir_Y, dir_Z
+
+    use m_ordering, only: get_index_dir, get_index_ijk
+
+#ifdef CUDA
+   use m_cuda_common, only: SZ
+   use m_cuda_tdsops, only: cuda_tdsops_t
+#else
+   use m_omp_common, only: SZ
+   use m_omp_backend, only: omp_backend_t
+#endif
+
+    implicit none
+
+    logical :: allpass = .true.
+    class(field_t), pointer :: u_x, u_y, u_z, u_x_original
+
+    integer :: nrank, nproc
+    integer :: ierr, i, j, k, i_, j_, k_, i__, j__, k__
```

Comment:
Could we use something more descriptive than `i_/i__` to represent the different index types, or atleast do
```
integer :: i_, j_, k_       ! Computed directional indices
integer :: i__, j__, k__ ! Computed Cartesian indices
```